restapi/src/main/java/io/stargate/web/docsapi/service/query/search/resolver/BaseResolver.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.web.docsapi.service.query.search.resolver;
+
+import com.bpodgursky.jbool_expressions.Expression;
+import com.bpodgursky.jbool_expressions.Literal;
+import io.stargate.web.docsapi.service.ExecutionContext;
+import io.stargate.web.docsapi.service.query.FilterExpression;
+import io.stargate.web.docsapi.service.query.search.resolver.impl.CnfResolver;
+import io.stargate.web.docsapi.service.query.search.resolver.impl.FullSearchResolver;
+import io.stargate.web.docsapi.service.query.search.resolver.impl.SingleExpressionResolver;
+import io.stargate.web.docsapi.service.query.search.weigth.ExpressionWeightResolver;
+
+/**
+ * Base resolver knows what {@link DocumentsResolver} should be created for the given {@link
+ * Expression}.
+ */
+public final class BaseResolver {
+
+  private BaseResolver() {}
+
+  /**
+   * Resolves the document resolver without any parent.
+   *
+   * @param expression {@link Expression}
+   * @param weightResolver {@link ExpressionWeightResolver}
+   * @return DocumentsResolver
+   */
+  public static DocumentsResolver resolve(
+      Expression<FilterExpression> expression,
+      ExpressionWeightResolver<FilterExpression> weightResolver,
+      ExecutionContext context) {
+    return resolve(expression, weightResolver, context, null);
+  }
+
+  /**
+   * Resolves the document resolver with optional parent.
+   *
+   * @param expression {@link Expression}
+   * @param weightResolver {@link ExpressionWeightResolver}
+   * @param parent parent or <code>null</code>
+   * @return DocumentsResolver
+   */
+  public static DocumentsResolver resolve(
+      Expression<FilterExpression> expression,
+      ExpressionWeightResolver<FilterExpression> weightResolver,
+      ExecutionContext context,
+      DocumentsResolver parent) {
+
+    // if we are hitting the literal TRUE, then return parent
+    // if no parent, than go for full search resolver
+    if (Literal.EXPR_TYPE.equals(expression.getExprType())) {
+      if (null != parent) {
+        return parent;
+      } else {
+        return new FullSearchResolver(context);
+      }
+    }
+
+    // if we have only one Filter expression, then to SingleExpression resolver
+    if (FilterExpression.EXPR_TYPE.equals(expression.getExprType())) {
+      return new SingleExpressionResolver((FilterExpression) expression, context, parent);
```

Comment:
No, because you can not have a single expression in an `And`, throws exception. Yes I know it's kind of a duplication, but once the single expression is left in a tree we must have smth like this....

Reply:
Not sure about the exception here... empty `AND` is certainly not allows, but one-element `AND` should be ok, I think. 

In any case this is mostly about consolidating resolution code. There is a more substantial discussion about that in another comment. Resolving for now.